index.js

```diff
@@ -97,25 +93,19 @@ var EventEmitter = {
      * @param {Function} [fn]
      */
     off: function (name, fn) {
-        if (!isArray(this._listeners)) {
-            this._listeners = [];
-        }
-
-        forEach(this._listeners, function (listener, index) {
-            if (name) {
-                if (listener.name === name) {
-                    if (isFunction(fn)) {
-                        if (listener.fn === fn) {
-                            this._listeners.splice(index, 1);
-                        }
-                    } else {
-                        this._listeners.splice(index, 1);
-                    }
-                }
+        this._listeners = !name ? [] : this._listeners.filter(function (listener, index) {
```

Comment:
I used this lib on platforms when is implemented ES3. `filter` method in available since ES5.
Please create helper for that.